analyzers/src/SonarAnalyzer.CSharp/Rules/ForLoopCounterChanged.cs

```diff
@@ -98,15 +99,13 @@ private static IEnumerable<ISymbol> LoopCounters(ForStatementSyntax node, Semant
         {
             var declaredVariables = node.Declaration == null
                 ? Enumerable.Empty<ISymbol>()
-                : node.Declaration.Variables
-                    .Select(v => semanticModel.GetDeclaredSymbol(v))
-                    .WhereNotNull();
+                : node.Declaration.Variables.Select(v => semanticModel.GetDeclaredSymbol(v));
 
-            var initializedVariables = node.Initializers
-                .Where(i => i.IsKind(SyntaxKind.SimpleAssignmentExpression))
-                .Select(i => semanticModel.GetSymbolInfo(((AssignmentExpressionSyntax)i).Left).Symbol);
+            var initializedVariables = node.Initializers.OfType<AssignmentExpressionSyntax>()
+                .SelectMany(x => x.AssignmentTargets())
+                .Select(x => semanticModel.GetSymbolInfo(x).Symbol ?? semanticModel.GetDeclaredSymbol(x));
```

Comment:
Doesn`t the type of the SyntaxNode define what method should be called to retrieve the symbol? It would be nicer to  called the correct one and not just "fall back" when one returns null.

Reply:
Isn't [this](https://github.com/SonarSource/sonar-dotnet/blob/29062274fa58a971eeb81ab04e0fd04bdfc882df/analyzers/src/SonarAnalyzer.CSharp/Rules/SetLocaleForDataTypes.cs#L76) the same logic you need here?

Reply:
Why do you think it is flawed? Can you come up with a case where it would not work?

Reply:
> Why do you think it is flawed?

I think `GetSymbolInfo` is in general the right thing to do and `GetDeclaredSymbol` is the exception. The example does it the other way around.

>  Can you come up with a case where it would not work?

No. I can't come up with an example. This is more like "This feels wrong to me".